sql/src/main/java/io/crate/protocols/postgres/RetryOnFailureResultReceiver.java

```diff
@@ -73,14 +88,36 @@ public void allFinished(boolean interrupted) {
     }
 
     @Override
-    public void fail(@Nonnull Throwable t) {
-        t = SQLExceptions.unwrap(t);
+    public void fail(@Nonnull Throwable wrappedError) {
+        final Throwable error = SQLExceptions.unwrap(wrappedError);
         if (attempt <= Constants.MAX_SHARD_MISSING_RETRIES &&
-            (SQLExceptions.isShardFailure(t) || t instanceof ConnectTransportException || indexWasTemporaryUnavailable(t))) {
-            attempt += 1;
-            retry();
+            (SQLExceptions.isShardFailure(error) || error instanceof ConnectTransportException || indexWasTemporaryUnavailable(error))) {
+
+            if (clusterService.state().blocks().hasGlobalBlock(RestStatus.SERVICE_UNAVAILABLE)) {
+                delegate.fail(error);
+            } else {
+                ClusterStateObserver clusterStateObserver =
+                    new ClusterStateObserver(initialState, clusterService, null, LOGGER, threadContext);
+                clusterStateObserver.waitForNextChange(new ClusterStateObserver.Listener() {
+                    @Override
+                    public void onNewClusterState(ClusterState state) {
+                        attempt += 1;
+                        retry();
+                    }
+
+                    @Override
+                    public void onClusterServiceClose() {
+                        delegate.fail(error);
```

Comment:
Is it ok that we call `delegate.fail(error)` with the unwrapped error now?
Why was this change needed?

Reply:
We already used the unwrapped error before. (`t = SQLExceptions.unwrap(t)`. I had to introduce a separate variable because it's now used in the anonymous Listener class which requires an effectively final variable.

Reply:
ok,Thx! somehow missed that... :-(